webapp/src/libs/services/BaseService.ts

```diff
@@ -47,15 +47,17 @@ export class BaseService {
             });
 
             if (!response.ok) {
-                const responseText = await response.text();
-
                 if (response.status === 504) {
                     throw Object.assign(new Error('The request timed out. Please try sending your message again.'));
                 }
 
-                const errorMessage = `${response.status}: ${response.statusText}${
-                    responseText ? ` => ${responseText}` : ''
-                }`;
+                const responseText = await response.text();
+                const responseDetails = responseText.split('--->');
```

Comment:
Is the `--->` a token defined somewhere else? I.e. will every error message contain it?

Reply:
The .NET Framework uses ---> to indicate the start of an inner exception stack trace. I can't find where this is documented but it's described in: https://learn.microsoft.com/en-us/dotnet/api/system.exception.tostring?view=net-7.0